generic-contracts/src/main/java/com/scalar/dl/genericcontracts/table/v1_0_0/Create.java

```diff
@@ -0,0 +1,82 @@
+package com.scalar.dl.genericcontracts.table.v1_0_0;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeType;
+import com.scalar.dl.ledger.contract.JacksonBasedContract;
+import com.scalar.dl.ledger.exception.ContractContextException;
+import com.scalar.dl.ledger.statemachine.Asset;
+import com.scalar.dl.ledger.statemachine.Ledger;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+public class Create extends JacksonBasedContract {
+
+  @Nullable
+  @Override
+  public JsonNode invoke(
+      Ledger<JsonNode> ledger, JsonNode arguments, @Nullable JsonNode properties) {
+
+    // Check the arguments
+    if (arguments.size() < 3
+        || arguments.size() > 4
+        || !arguments.has(Constants.TABLE_NAME)
+        || !arguments.get(Constants.TABLE_NAME).isTextual()
+        || !arguments.has(Constants.TABLE_KEY)
+        || !arguments.get(Constants.TABLE_KEY).isTextual()
+        || !arguments.has(Constants.TABLE_KEY_TYPE)
+        || !arguments.get(Constants.TABLE_KEY_TYPE).isTextual()) {
+      throw new ContractContextException(Constants.INVALID_TABLE_FORMAT);
+    }
+
+    if (!isSupportedKeyType(arguments.get(Constants.TABLE_KEY_TYPE).asText())) {
+      throw new ContractContextException(Constants.INVALID_KEY_TYPE);
+    }
+
+    if (arguments.has(Constants.TABLE_INDEXES)) {
+      validateIndexes(arguments.get(Constants.TABLE_INDEXES));
+    }
+
+    // Get the table existence
+    String assetId = getAssetIdForTable(arguments.get(Constants.TABLE_NAME).asText());
+    Optional<Asset<JsonNode>> asset = ledger.get(assetId);
+    if (asset.isPresent()) {
+      throw new ContractContextException(Constants.TABLE_ALREADY_EXISTS);
+    }
+
+    // Put the asset records
+    ledger.put(Constants.ASSET_ID_METADATA_TABLES, arguments);
+    ledger.put(assetId, arguments);
```

Comment:
We need to manage an index `metadata_tables` to track all existing tables for the listing contract while keeping the efficiency (a point query) when getting the table information.